extensions-core/s3-extensions/src/main/java/io/druid/storage/s3/KmsServerSideEncryption.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.druid.storage.s3;
+
+import com.amazonaws.services.s3.model.CopyObjectRequest;
+import com.amazonaws.services.s3.model.PutObjectRequest;
+import com.amazonaws.services.s3.model.SSEAwsKeyManagementParams;
+
+import javax.annotation.Nullable;
+
+class KmsServerSideEncryption implements ServerSideEncryption
+{
+  static final String NAME = "kms";
+
+  @Nullable
+  private final String keyId;
+
+  KmsServerSideEncryption(@Nullable String keyId)
```

Comment:
This does not allow multiple region keys I don't think. That's probably ok here though since you're doing server side encryption.

Reply:
That's a good point. I'll file an issue about this.